dune-precice/dune-precice/couplinginterface.hh

```diff
@@ -134,26 +133,28 @@ namespace Dune::preCICE {
     , mesh_name_(parameters.mesh_name)
     , read_data_name_(parameters.read_data_name)
     , write_data_name_(parameters.write_data_name)
-  {}
+  {
+    mesh_name_view = precice::string_view(mesh_name_);
+    read_data_name_view = precice::string_view(read_data_name_);
+    write_data_name_view = precice::string_view(write_data_name_);
+  }
     
   template <int dim, typename VectorType, typename ParameterClass>
   template <typename PolynomialBasis>
-  double CouplingInterface<dim, VectorType, ParameterClass>::initialize(
+  void CouplingInterface<dim, VectorType, ParameterClass>::initialize(
     const PolynomialBasis &polynomial_basis) 
   {
-    mesh_id_       = precice.getMeshID(mesh_name_);
-    read_data_id_  = precice.getDataID(read_data_name_, mesh_id_);
-    write_data_id_ = precice.getDataID(write_data_name_, mesh_id_);
-            
     VectorType coordinates_of_boundary;
     auto coordinate = [] (auto x) { return x; };
  
     interpolate(polynomial_basis, coordinates_of_boundary, coordinate, is_coupling_boundary_);
 
     std::vector<double> interface_nodes_positions;
+    const precice::span<double> interface_nodes_positions_span = precice::span(interface_nodes_positions);
     for(int i=0; i<coordinates_of_boundary.N(); i++) {
       for(int j=0; j<dim; j++) {
-        if(is_coupling_boundary_[i][j]) interface_nodes_positions.push_back(coordinates_of_boundary[i][j]);
+        if(is_coupling_boundary_[i][j]) {interface_nodes_positions.push_back(coordinates_of_boundary[i][j]);
+        interface_nodes_ids_.push_back(j*coordinates_of_boundary.N()+i);}
```

Comment:
This syntax is difficult to read. It would be better to have the operations inside the if loop on separate lines.